BGL/include/CGAL/boost/graph/IO/STL.h

```diff
@@ -238,13 +243,13 @@ bool write_STL(std::ostream& os,
       Point_ref r = get(vpm, source(h, g));
       Vector n = collinear(p, q, r) ? Vector(1, 0, 0) : unit_normal(p, q, r);
 
-      os << "facet normal " << n << "\nouter loop\n";
-      os << "vertex " << p << "\n";
-      os << "vertex " << q << "\n";
-      os << "vertex " << r << "\n";
-      os << "endloop\nendfacet\n";
+      os << "facet normal " << n << "\nouter loop"<<std::endl;;
+      os << "vertex " << p <<std::endl;
+      os << "vertex " << q <<std::endl;
+      os << "vertex " << r <<std::endl;
+      os << "endloop\nendfacet"<<std::endl;
```

Comment:
I wonder why you insist on turning `\n` into `std::endl`. You should read [the documentation of `std::endl`](https://en.cppreference.com/w/cpp/io/manip/endl).

Reply:
With the \n, after around 14K lines, the stream seems to be full or something, and the end of the file is not written. endl flushes the stream, that's what I was going for. And indeed, I don't have the writing bug after I turned the \n into endl. But if you really prefer \n, I can absolutely only keep the last endl and turned back to \n for the rest.

Reply:
You could keep one `std::endl` (the last one) for each iteration in the loop. That will already save a lot of `std::flush`.